### PR TITLE
Create data model

### PR DESCRIPTION
**Pull request by taldr27**
In this milestone I:
- Added linters files including `rubocop` and `styleint`.
- Added the references in each migrate file.
- Generated the new models.
- Generated the migration files for user, comment, post and likes.
- Used `db:migrate` to migrate all the migration files.
gif

### CODE REVIEW

File: db/migrate/20230124144929_create_comments.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.references :user
```
Comment (omar-muhamad) [inline db/migrate/20230124144929_create_comments.rb]: - [x] As per this [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_1.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram), Kindly create the foreign key between users table and comments table as `author_id` not `user_id` like you did here.
[Hint]: you can do it like that:
```ruby
t.references :author, foreign_key: { to_table: 'users' }
```

File: db/migrate/20230124144934_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+
+      t.timestamps
```
Comment (omar-muhamad) [inline db/migrate/20230124144934_create_posts.rb]: - [x] Kindly create the foreign key between users table and posts table as `author_id`.
- [x] Please add `title`, `text`, `comments_counter`, and `likes_counter` columns as mentioned in the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

File: db/migrate/20230124144937_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+
+      t.timestamps
```
Comment (omar-muhamad) [inline db/migrate/20230124144937_create_likes.rb]: - [x] Kindly create references here as foreign keys between users table and likes table as `author_id`.
- [x] Kindly create references here as foreign keys between posts table and likes table.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @taldr27,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] I can see that you created all your migrations using `rails g scaffold` so it created models and tests too and that's not required now, So please delete these models and tests you will implement them later and leave only the migration files.
[Hint] you can create these files using the `rails generate migration` command see [this link](https://guides.rubyonrails.org/active_record_migrations.html#creating-a-migration) for more info
 
- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by omar-muhamad** [comment]
Hi @taldr27,
After you ask for a review to your project,
Kindly don't push any changes until you get the review from the code reviewer. because you added 3 commits that I didn't see until I submitted my review.
Have a great day 😊

**Comment by taldr27** [comment]
Hello @Omar-Muhamad,
Thanks for your review, I'm sorry I forgot to submit some of my changes that's why I pushed after asking for a review, I won't do that again. 🙏

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @taldr27 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️

- [ ] As requested by the previous reviewer, Please delete the models and tests that are created by the `rails g scaffold` command, you will implement them later and leave only the migration files.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BertrandConxy — APPROVED** [review]
Hi @taldr27, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230124144929_create_comments.rb (hunk 1 above)
Comment (taldr27) [reply db/migrate/20230124144929_create_comments.rb]: I just updated that `t.references` line and I used your suggestion in my file, thank you.

File: db/migrate/20230124144934_create_posts.rb (hunk 2 above)
Comment (taldr27) [reply db/migrate/20230124144934_create_posts.rb]: I just updated that `t.references` line with the `author_id`, I also forgot to push my changes in that table, it's done now, thank you.

**Comment by taldr27** [comment]
@thecodechaser , @BertrandConxy thank you very much for your reviews and feedback 👍.

**Comment by BertrandConxy** [comment]
you are welcome

**Pull request merged**